includes/Core/Authentication/Setup.php

```diff
@@ -260,7 +260,7 @@ public function handle_action_exchange_site_code() {
 	 */
 	protected function verify_nonce( $nonce, $action = Google_Proxy::NONCE_ACTION ) {
 		if ( ! wp_verify_nonce( $nonce, $action ) ) {
-			Authentication::invalid_nonce_error( $action );
+			$this->authentication->invalid_nonce_error( $action );
```

Comment:
Since we are touching other places of the code rather that what's defined in the ACs, can you add a couple of points in the AC to check Both Setup and Reset functionality with and without the token expiration snippet to make sure everything is working as intended?

Reply:
Updated.